src/resolver/mod.rs

```diff
@@ -71,17 +74,24 @@ impl Resolver {
     }
 }
 
-fn resolve(
+async fn resolve(
```

Comment:
`async` is more complex than a plain old synchronous method execution. It should only be used if some benefits justify this additional complexity.

Here `resolve` has all data available in memory that it needs to do it's job. There shouldn't be the need to wait for anything in the `Future`. Therefore this method should stay synchronous.